doc/making-a-visualizer.md

```diff
@@ -126,31 +138,63 @@ running in, or access to its document object model (DOM), you shouldn't need
-from frame to frame.
+There are two places to do pre-computation before the sketch begins looping
+through frames of the drawing. The first is `presketch()`, which runs
+asynchronously, meaning that the browser will not be blocked while this
+function completes. This is not a part of p5.js, but a part of Numberscope.
+
+If you implement `presketch()`, begin by calling `super.presketch()`, which
+will initialize the sequence. After this call, you have access to the
+sequence, so you can do sequence-dependent validation and initialization here.
+You do have the opportunity to set up private state variables. For example,
+this is a good place to populate an array with time-consuming precomputed
+values you will use repeatedly during the sketch. However, you still have no
+access to the p5 canvas.
+
+The next opportunity is when the p5 graphics context becomes available, which
+occurs after `presketch()` completes. At this time, `setup()` is called,
+giving you your first chance to set graphics options and draw on the canvas.
+For example, p5 colors cannot be dealt with in `presketch()`, but can be set
+up here. This is a good place for one-time graphics operations, like painting
+the background of a static visualizer or setting colors and stroke options
+that won't change from frame to frame.
 
 If you implement `setup()`, start by calling `super.setup()`, which includes
 the [`createCanvas()`](https://p5js.org/reference/#/p5/createCanvas) call that
 must appear in every p5 setup function.
 
-When `setup()` is called, the visualizer will always be attached to a
-sequence, so you can do sequence-dependent validation and initialization here.
+When a visualizer is resized, or the restart button on Numberscope is pressed,
+the class function `reset()` is called. By default, this calls both
+`presketch()` and `setup()`, and a new canvas is created. However, the
+visualizer object constructor is not re-run and any data stored in variables
+in the visualizer object persists. That means that you have the option to
+forgo re-doing expensive pre-computations, by overriding `reset()` from the
+`P5Visualizer` base class.
 
 -   **p5 Template:** Go to the beginning of the sequence. Create palette
     colors. Set text alignment.
--   **Differences:** Fill in the default value of the `levels` property if it
-    was left unset in the parameters dialog. Do a consistency check between
-    the sequence being visualized and the parameters.
 
 #### 🔩️ Show or stop the visualization; depart from a page element _(advanced)_
 
-You shouldn't need to implement `show()`, `stop()`, or `depart()`. You can
-learn about them from the [visualizer interface](behind-the-scenes.md)
+You shouldn't frequently need to implement `show()`, `stop()`, or `depart()`.
+You can learn about them from the [visualizer interface](behind-the-scenes.md)
 documentation, and from how they're implemented in the `P5Visualizer` base
```

Comment:
did you want to review and update that behind-the-scenes document as well in this same PR? Or leave it for another PR? I am sure that it is also out of date in various ways.

Reply:
If we can leave it for another PR, I'd prefer that (just don't want to dive into it right now).  I'll add to the post-delft todo so we can make an issue...

Reply:
Yup, fine to defer. Resolving.